### ISSUE TITLE
[HOLD] [$500] IOS - Bank account - Keyboard pulls up when redirected to BA page after tapping Pay with Expensify

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#32116]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.4.0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause - Internal Team
**Slack conversation:**

## Action Performed:
Precondition: The default currency for the workspace is USD.
1. Go to the workspace chat from the precondition.
2. Create a workspace expense (Manual).
3. Tap on the composer.
4. Tap Pay with Expensify.
5. Tap Business bank account.

## Expected Result:
Describe what you think should've happened

## Actual Result:
Keyboard pulls up.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [x] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/93399543/34f245b2-5bc2-419c-8ba2-bf6f4278138a

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01e85396f1cd4ec884</li>
        <li>Upwork Job ID: 1729490424428687360</li>
        <li>Last Price Increase: 2023-12-05</li>
    </ul>
</details>

**Comment by situchan** [comment]
Makes sense to hold

**Comment by situchan** [comment]
Not overdue

**Comment by situchan** [comment]
Not overdue

**Comment by MitchExpensify** [comment]
Switching to Weekly while on hold

**Comment by MitchExpensify** [comment]
On hold https://github.com/Expensify/App/issues/24452#issuecomment-1712021524

**Comment by MitchExpensify** [comment]
On hold

**Comment by MitchExpensify** [comment]
On hold

**Comment by situchan** [comment]
still holding

**Comment by situchan** [comment]
Same

**Comment by MitchExpensify** [comment]
HOLDING THE LINE

**Comment by MitchExpensify** [comment]
On hold https://github.com/Expensify/App/issues/24452#issuecomment-1712021524

**Comment by MitchExpensify** [comment]
On hold https://github.com/Expensify/App/issues/24452#issuecomment-1712021524

**Comment by MitchExpensify** [comment]
Touches paying with a business bank account so wave collect seems most appropriate here

**Comment by MitchExpensify** [comment]
On hold https://github.com/Expensify/App/issues/24452#issuecomment-1712021524

**Comment by MitchExpensify** [comment]
On hold https://github.com/Expensify/App/issues/24452#issuecomment-1712021524

**Comment by MitchExpensify** [comment]
On hold https://github.com/Expensify/App/issues/24452#issuecomment-1712021524

**Comment by MitchExpensify** [comment]
On hold https://github.com/Expensify/App/issues/24452#issuecomment-1712021524

**Comment by MitchExpensify** [comment]
On hold https://github.com/Expensify/App/issues/24452#issuecomment-1712021524

**Comment by MitchExpensify** [comment]
On hold https://github.com/Expensify/App/issues/24452#issuecomment-1712021524

**Comment by MitchExpensify** [comment]
On hold https://github.com/Expensify/App/issues/24452#issuecomment-1712021524

**Comment by situchan** [comment]
I think we can remove hold.

**Comment by situchan** [comment]
This is no longer on hold.
Just retested. Keyboard still briefly shows, though dismissed finally.


https://github.com/user-attachments/assets/d659328e-ba46-4b0b-8098-8ed4cea7c24d

**Comment by MitchExpensify** [comment]
Reassigning while I'm out 🙇

**Comment by situchan** [comment]
> That sounds good! As far as I can tell, this is no longer on hold. Is that right, @situchan?

right. The bug is https://github.com/Expensify/App/issues/32116#issuecomment-2223685912 and waiting for proposals

**Comment by johncschuster** [comment]
Great! Let's drum up some proposals then!

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The keyboard pulls up again when redirecting to the BA page after selecting the pay option from a Modal.

### What is the root cause of that problem?
In iOS, by default, the app will refocus on the input if the modal is closed if the input is previously focused.

Focus composer -> open payment method modal -> select payment method -> modal closes -> composer is refocused -> BA page opens

### What changes do you think we should make in order to solve the problem?

In https://github.com/Expensify/App/issues/24452, we introduce a new focus manager that handles the focus and refocus manually which gives us more flexibility compared to the OS default behavior. To fix this specific issue, we need to add 2 things.

First, we need to enable the new focus manager by passing the `shouldEnableNewFocusManagement` props to the ([AddPaymentMethodMenu](https://github.com/Expensify/App/blob/54fdb24587f6944941b69610a908878d5861e049/src/components/AddPaymentMethodMenu.tsx#L73-L111)) modal.

Enabling the new focus manager will save the focused input when a modal is shown,
https://github.com/Expensify/App/blob/54fdb24587f6944941b69610a908878d5861e049/src/components/Modal/BaseModal.tsx#L68-L73

and restoring the focus when it closes.
https://github.com/Expensify/App/blob/54fdb24587f6944941b69610a908878d5861e049/src/components/Modal/BaseModal.tsx#L90-L91

However, we don't want to refocus when we select a payment method. To achieve that, we need to pass an additional prop to the (AddPaymentMethodMenu) modal called `restoreFocusType`.
```
const [restoreFocusType, setRestoreFocusType] = useState(undefined);
```

If we are selecting a payment method, we will set the `restoreFocusType` to DELETE (no refocus), 
https://github.com/Expensify/App/blob/54fdb24587f6944941b69610a908878d5861e049/src/components/AddPaymentMethodMenu.tsx#L80

```
onItemSelected={() => {
    setRestoreFocusType(CONST.MODAL.RESTORE_FOCUS_TYPE.DELETE);
    onClose();
}}
```

otherwise, it will be undefined/DEFAULT (refocus).
https://github.com/Expensify/App/blob/54fdb24587f6944941b69610a908878d5861e049/src/components/AddPaymentMethodMenu.tsx#L76

```
onClose={() => {
    setRestoreFocusType(undefined);
    onClose();
}}
```

**Comment by johncschuster** [comment]
@situchan can you review @bernhardoj's proposal above?

**Comment by situchan** [comment]
@bernhardoj thanks for the proposal. It looks promising. Do you mind sharing test branch?

**Comment by situchan** [comment]
bump

**Comment by bernhardoj** [comment]
Sorry for the delay, here is the test branch: https://github.com/bernhardoj/Expensify/tree/fix/32116-keyboard-shows-up-when-open-ba-page

**Comment by johncschuster** [comment]
Thanks, @bernhardoj!

**Comment by situchan** [comment]
@bernhardoj's [proposal](https://github.com/Expensify/App/issues/32116#issuecomment-2238966052) looks good to me.
🎀👀🎀 C+ reviewed

**Comment by rlinoz** [comment]
Thanks, assigning @bernhardoj

**Comment by johncschuster** [comment]
@situchan can we get a status on the review?

**Comment by situchan** [comment]
> @situchan can we get a status on the review?

C+ approved

**Comment by johncschuster** [comment]
Looks like the PR above was deployed to staging 👍

**Comment by johncschuster** [comment]
Acknowledged! I'll issue payment after the regression threshold is cleared.

**Comment by johncschuster** [comment]
Bumping this to reset the Melvin clock. I will pay this out on the 19th assuming no regressions.

**Comment by johncschuster** [comment]
Payment Summary:

Contributor: @bernhardoj paid $500 via NewDot
Contributor+: @situchan owed $500 via [Upwork](https://www.upwork.com/nx/wm/offer/103325898) - PAID 🎉

**Comment by johncschuster** [comment]
@situchan, do you feel we need a regression test? If so, can you provide one?

**Comment by situchan** [comment]
I don't think we need regression test as this is iOS only issue and does not affect product.

**Comment by bernhardoj** [comment]
Requested in ND.

**Comment by johncschuster** [comment]
> I don't think we need regression test as this is iOS only issue and does not affect product.

Awesome. In that case, I'll close this up since we don't need to keep it open for ND payments. Thank you both for your contributions!

**Comment by JmillsExpensify** [comment]
$500 approved for @bernhardoj
